srml/session/src/lib.rs

```diff
@@ -563,9 +584,19 @@ mod tests {
 			assert_ok!(Session::set_keys(Origin::signed(2), UintAuthorityId(5), vec![]));
 			assert_eq!(authorities(), vec![UintAuthorityId(1), UintAuthorityId(2), UintAuthorityId(3)]);
 
-			// Block 4: Session rollover, authority 2 changes.
+			// Block 4: Session rollover; no visible change.
 			System::set_block_number(4);
 			Session::on_initialize(4);
+			assert_eq!(authorities(), vec![UintAuthorityId(1), UintAuthorityId(2), UintAuthorityId(3)]);
+
+			// Block 5: No change.
+			System::set_block_number(5);
```

Comment:
Recommendation: use some helpers like this for set+finalize operations
https://github.com/paritytech/substrate/blob/master/srml/staking/src/mock.rs#L294